pkg/sql/sqlbase/fk.go

```diff
@@ -329,6 +329,10 @@ func (f *fkBatchChecker) addCheck(row tree.Datums, source *baseFKHelper) error {
 func (f *fkBatchChecker) runCheck(
 	ctx context.Context, oldRow tree.Datums, newRow tree.Datums,
 ) error {
+
+	if f == nil {
```

Comment:
yeah that's a "weird fix" all right.

Can you find out when `f` can be nil and why and see whether you could avoid the `runCheck` call altogether in that case.

Reply:
What would probably work is to re-add the checks I removed because of Bram's review comment earlier, checking for whether the number of foreign keys to check is 0 (scroll up in this CR and look for `if len(ru.Fks.inbound.fks) == 0 && len(ru.Fks.outbound.fks) == 0`).